kafkabp/group_consumer_test.go

```diff
@@ -0,0 +1,169 @@
+package kafkabp
+
+import (
+	"context"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/Shopify/sarama"
+)
+
+func TestGroupConsumer(t *testing.T) {
+	// mock producer side by injecting a mocked consumer group
+	mockProducer := newMockConsumerGroup()
+
+	// test implemenation of a consumer
+	testConsumer := newTestConsumer()
+
+	// inject mocked consumer group
+	gc := &groupConsumer{
+		consumer: mockProducer,
+	}
+
+	// start group consumer in the background and collect all errors
+	errors := make(chan error)
+	go func() {
+		errors <- gc.Consume(testConsumer.Consume, testConsumer.Error)
+	}()
+
+	// expect two messages
+	testConsumer.expectedMessages = 2
+
+	// producer will delivery messages to test consumer
+	mockProducer.messages <- &sarama.ConsumerMessage{
+		Value: []byte(`{"data":"a"}`),
+	}
+	mockProducer.messages <- &sarama.ConsumerMessage{
+		Value: []byte(`{"data":"b"}`),
+	}
+	mockProducer.claimsAssigned <- true
+
+	// wait for processing to finish but fail if it blocks too long
+	select {
+	case <-testConsumer.processed:
+	case <-time.After(1 * time.Second):
```

Comment:
🔕 I would prefer a lower timeout for unit tests, mainly for making sure that running the unit test won't take too long. But I understand that using a lower timeout has the risk of making the tests more flaky.